### PR TITLE
Implementing suggestions for search

### LINKED ISSUE #429: Implement "did you mean. . ." corrections in /v1/images endpoint
Return a `suggestions` field in queries to `/v1/images`.

Example request and response:
`GET /v1/images?dags`
```
{
  "suggestions": "dogs" 
   "result_count" : 10000,
   "results" : [
      {
         "license_version" : "2.0",
         "foreign_landing_url" : "https://www.flickr.com/photos/48889073020@N01/3042355",
         "title" : "Big Dog, Little Dog",
         "detail_url" : "https://api.creativecommons.engineering/v1/images/3474484f-f75c-45bd-85f7-0efa8678bc1a",
         "width" : 500,
         "height" : 375,
         "url" : "https://farm1.staticflickr.com/3/3042355_c700071469.jpg",
         "thumbnail" : "https://farm1.staticflickr.com/3/3042355_c700071469_m.jpg",
         "related_url" : "https://api.creativecommons.engineering/v1/recommendations/images/3474484f-f75c-45bd-85f7-0efa8678bc1a",
         "license_url" : "https://creativecommons.org/licenses/by-nc-nd/2.0/",
         "source" : "flickr",
         "fields_matched" : [
            "description",
            "tags.name",
            "title"
         ],
         "id" : "3474484f-f75c-45bd-85f7-0efa8678bc1a",
         "tags" : [
            {
               "name" : "molly"
            },
. . .
```

The following modifications will need to be made:
- Update `search_controller` to use `suggesters`. See the [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters.html) for information on how this feature works.
- Update the Image serializer to return the `suggestions` field.

Whether the search should be repeated again with the suggested term is left at the discretion of the client; we should only return the suggestions and not repeat the search.

We will use this to implement spelling corrections on the front end.
Autocomplete

### PR DESCRIPTION
**Pull request by AyanChoudhary**
Fixes #429 

This pull request addresses the requirement of a suggestion field in the result image search to facilitate the new feature wherein the handling of misspelled searches can be done more efficiently.

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by AyanChoudhary** [comment]
I was just going through ci tests log, initially it looked like it was due to linting errors, but since I have fixed it doesn't seem to be the case now. Can you help me in figuring out the exact reason for this failing test

**Comment by AyanChoudhary** [comment]
@aldenstpage All the builds are now passing, please review.

File: cccatalog-api/cccatalog/api/views/image_views.py [hunk 1]
```diff
@@ -75,7 +76,10 @@ def get(self, request, format=None):
 
         search_index = 'search-qa' if qa else 'image'
         try:
-            results, page_count, result_count = search_controller.search(
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/image_views.py]: Could you undo the style changes in this file on lines 79-82 and 254-255? After that, I'll merge this PR. It looks good!

Comment (AyanChoudhary) [reply cccatalog-api/cccatalog/api/views/image_views.py]: I did not do any changes on lines 245-255, but I have reverted those on 79-82 in the latest commit

**Comment by AyanChoudhary** [comment]
@aldenstpage the build is no failing due to style changes with the command violation
pycodestyle cccatalog-api/cccatalog --exclude='cccatalog-api/cccatalog/api/migrations' --max-line length=80 --ignore=E402
So I guess we will have to work with the previous style.

File: cccatalog-api/cccatalog/api/views/image_views.py [hunk 2]
```diff
@@ -246,7 +248,8 @@ def get(self, request, identifier, format=None):
                 with_xmp = ccrel.embed_xmp_bytes(img_bytes, work_properties)
                 return FileResponse(with_xmp, content_type='image/jpeg')
             except (libxmp.XMPError, AttributeError) as e:
-                # Just send the EXIF-ified file if libxmp fails to add metadata.
+                # Just send the EXIF-ified file
```
Comment (aldenstpage) [inline cccatalog-api/cccatalog/api/views/image_views.py]: There's a new line break here, this is what I was referring to

**Comment by aldenstpage** [comment]
Ah, I see there's another keyword there making it run over the line limit, so there will need to be a change there after all. I'm going to make a few whitespace tweaks and merge this.

**Pull request merged**